### PR TITLE
Add eval cli

### PR DESCRIPTION
**Pull request by thomasyu888**
Remove `cli` functionality from inside the module, and add to `click` group

Usage: 

```
nlp-cli evaluate prediction --pred_filepath tests/data/prediction_1.json --gold_filepath tests/data/goldstandard_1.json
```

### CODE REVIEW

**Comment by tschaffter** [comment]
@yy6linda Is this PR ready for review (it's marked as a Draft)

**Comment by thomasyu888** [comment]
I will give it a review as well tomorrow.

File: nlpsandboxclient/evaluation.py [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+# physical address
+from abc import ABCMeta
+import json
+import re
+
+
+# take as input the location of
+class Evaluation(metaclass=ABCMeta):
+    """Evaluate the different types"""
+    gs_dict_seq = dict()
```
Comment (tschaffter) [inline nlpsandboxclient/evaluation.py:14]: General comments about variable names:
- do not use abbreviations whenever possible to make the code more readable. Here an external developer would be confused with name like "gs" and "sys" at the time of the first reading.
- do not include the type of the variable in the variable name (shorten the variable name)

**Review by tschaffter — APPROVED** [review]
I have started to define more formally the evaluation object in the LucidChart document below:
https://lucid.app/lucidchart/670f484d-4a3f-42d8-bdcd-7aa34dce783e

This code will be refactored once we settle on the evaluation object schemas. This work will also be useful for the Challenge Platform, which is why I'm interested in doing this work as part of this project.

**Comment by thomasyu888** [comment]
I think this is a good idea, but just by looking at this, I think this challenge has a custom Evaluation object.  If you think about other challenges, the object could probably have:

* goldstandard_path
* prediction_path
* metrics (as an array)

The Evaluation object could probably be part of a Queue object, as this is how we could link a Submission to its Evaluation.

**Comment by tschaffter** [comment]
My take is that we can go ahead with the design implemented in this PR so that we can do early testing of the infra, then refactor the code once we have identified a more efficient design.

**Pull request merged**